BCnEnc.Net/Shared/DdsFile.cs

```diff
@@ -34,60 +34,56 @@ public static DdsFile Load(Stream s)
 					throw new FormatException("The file does not contain a dds file.");
 				}
 				var header = br.ReadStruct<DdsHeader>();
-				DdsHeaderDxt10 dxt10Header = default;
+				DdsHeaderDx10 dx10Header = default;
 				if (header.dwSize != 124)
 				{
 					throw new FormatException("The file header contains invalid dwSize.");
 				}
 
-				var dxt10Format = header.ddsPixelFormat.IsDxt10Format;
+				var dx10Format = header.ddsPixelFormat.IsDxt10Format;
 
 				DdsFile output;
 
-				if (dxt10Format)
+				if (dx10Format)
 				{
-					dxt10Header = br.ReadStruct<DdsHeaderDxt10>();
-					output = new DdsFile(header, dxt10Header);
+					dx10Header = br.ReadStruct<DdsHeaderDx10>();
+					output = new DdsFile(header, dx10Header);
 				}
 				else
 				{
 					output = new DdsFile(header);
 				}
 
 				var mipMapCount = (header.dwCaps & HeaderCaps.DdscapsMipmap) != 0 ? header.dwMipMapCount : 1;
-				var faceCount = (header.dwCaps2 & HeaderCaps2.Ddscaps2Cubemap) != 0 ? (uint)6 : (uint)1;
+				var faceCount = (header.dwCaps2 & HeaderCaps2.Ddscaps2Cubemap) != 0 ? 6u : 1u;
 				var width = header.dwWidth;
 				var height = header.dwHeight;
 
 				for (var face = 0; face < faceCount; face++)
 				{
-					var sizeInBytes = Math.Max(1, (width + 3) / 4) * Math.Max(1, (height + 3) / 4);
-					if (!dxt10Format)
+					var format = dx10Format ? dx10Header.dxgiFormat : header.ddsPixelFormat.DxgiFormat;
+
+					uint sizeInBytes;
+					if (format.IsCompressedFormat())
 					{
-						if (header.ddsPixelFormat.IsDxt1To5CompressedFormat)
+						sizeInBytes = (uint)Math.Max(1, ((width + 3) & ~3) >> 2) * (uint)Math.Max(1, ((height + 3) & ~3) >> 2);
+						if (format == DxgiFormat.DxgiFormatBc1Unorm ||
+							format == DxgiFormat.DxgiFormatBc1UnormSrgb ||
+							format == DxgiFormat.DxgiFormatBc1Typeless)
```

Comment:
I would put this in a separate function since the mips need to do a similar calculation at around line 100.

Reply:
Done, and replaced the line 100 occurence with this method as well.